src/main/scala/com/scality/clueso/SparkUtils.scala

```diff
@@ -0,0 +1,129 @@
+package com.scality.clueso
+
+import java.io.{File, IOException}
+import java.net.URI
+
+import com.scality.clueso.query.{MetadataQuery, MetadataQueryExecutor}
+import com.typesafe.config.ConfigFactory
+import com.typesafe.scalalogging.LazyLogging
+import org.apache.hadoop.conf.{Configuration => HadoopConfig}
+import org.apache.hadoop.fs.{FileStatus, FileSystem, Path, PathFilter}
+import org.apache.spark.sql.functions.{col, lit}
+import org.apache.spark.sql.{Row, SparkSession}
+
+
+object SparkUtils extends LazyLogging {
+  def loadCluesoConfig(confFilePath: String) = {
+    val parsedConfig = ConfigFactory.parseFile(new File(confFilePath))
+    val _config = ConfigFactory.load(parsedConfig)
+
+    new CluesoConfig(_config)
+  }
+
+  def buildHadoopFs(config: CluesoConfig) = {
+    FileSystem.get(new URI(PathUtils.landingURI(config)), hadoopConfig(config))
+  }
+
+  def hadoopConfig(config: CluesoConfig): HadoopConfig = {
+    val c = new HadoopConfig()
+    c.set("fs.s3a.connection.ssl.enabled", config.s3SslEnabled)
+    c.set("fs.s3a.endpoint", config.s3Endpoint)
+    c.set("fs.s3a.impl", "org.apache.hadoop.fs.s3a.S3AFileSystem")
+    c.set("fs.s3a.access.key", config.s3AccessKey)
+    c.set("fs.s3a.secret.key", config.s3SecretKey)
+    c.set("fs.s3a.path.style.access", config.s3PathStyleAccess)
+    c.set("spark.serializer", "org.apache.spark.serializer.KryoSerializer")
+
+    c
+  }
+
+  def buildSparkSession(config : CluesoConfig) = {
+    SparkSession
+      .builder
+      .config("spark.sql.parquet.cacheMetadata", "false")
+      .config("spark.hadoop.fs.s3a.fast.upload", "true")
```

Comment:
shouldn't all the fs.s3a stuff be in the hadoop config?

Reply:
for fast.upload it seems unlikely we need this since our files are so small and our s3 is local. what have you seen as the largest file we will write? (https://hadoop.apache.org/docs/r2.8.0/hadoop-aws/tools/hadoop-aws/index.html#Stabilizing:_S3A_Fast_Upload) If I am reading correctly, by using a path to be our buffer location, we could end up writing to disk twice -- once in ./tmp and then in our s3.

Reply:
the largest.. probably 70-100MB. do you think it's worth s3a fast upload?